src/components/Sidebars/Payback.js

```diff
@@ -123,22 +123,46 @@ export default Payback;
 const checkForProxyAndAllowance = async (
   maker,
   updateState,
-  setHasAllowance
+  setHasAllowance,
+  madeInitialCheck
 ) => {
   const proxyAddress = await maker.service('proxy').getProxyAddress();
   const token = maker.getToken('MDAI');
-  const hasAllowance =
-    proxyAddress &&
+
+  const checkAllowance = async () =>
+    !!proxyAddress &&
     (await token.allowance(maker.currentAddress(), proxyAddress)).eq(
       MAX_UINT_BN
     );
 
+  const pollCheckAllowance = async () => {
+    let numTries = 5;
+    let allowance;
+    while (numTries > 0) {
+      numTries--;
+      allowance = await checkAllowance();
+      if (allowance) return allowance;
+    }
+    return false;
+  };
+
+  let hasAllowance = await checkAllowance();
+  if (madeInitialCheck && !hasAllowance) {
+    hasAllowance = await pollCheckAllowance();
+  }
```

Comment:
This could be reverted with:
```javascript
let hasAllowance = await checkAllowance();
```
but I found that sometimes`checkAllowance()` would return false for roughly 1/10 attempts when called directly after awaiting on the transaction on kovan (using infura). This would lead me to assume that we are running into the separate nodes issue possibly. Let me know if this has to be reverted